PyInstaller/building/osx.py

```diff
@@ -192,7 +192,12 @@ def assemble(self):
                     strict_arch_validation=(typ == 'EXTENSION'),
                 )
             # Add most data files to a list for symlinking later.
-            if typ == 'DATA' and base_path not in _QT_BASE_PATH:
+            # Exempt python source files from this relocation, because their real path might need to resolve
+            # to the directory that also contains the extension module.
+            relocate_file = type == 'DATA' and base_path not in _QT_BASE_PATH
```

Comment:
@rokm @bwoodsend A test is definitely needed as nothing caught the typo here `type` should be `typ` which means that no files are relocated ever

Reply:
Indeed. The effect of this error should be caught by the metadata collection test that will be added in #7187. 

Thanks for investigating and fixing the regression!